File: 02_rookie/week6/1_two-sum_watermelon.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    // 한 번 검사한 숫자에 대하여 숫자-인덱스 쌍을 저장할 객체
    const checked = {};
    
    for (const [index, num] of nums.entries()) {
        if (target - num in checked) {
            return [index, checked[target - num]];
        }
        checked[num] = index;
    }
};

File: 02_rookie/week6/1_two-sum_watermelon.go
func twoSum(nums []int, target int) []int {
    var result []int
    checked := make(map[int]int)
    
    for index, num := range nums {
        if targetIndex, ok := checked[target - num]; ok {
            result = []int{index, targetIndex}
        }
        checked[num] = index
    }
    return result
}
